### PR TITLE
Don't use docker shim if only using a mounted docker.sock instead of docker-in-docker

### PR DESCRIPTION
**Pull request by Hoverbear**
##### Description

I believe this fixes https://github.com/DeterminateSystems/nix-installer-action/issues/65.

##### Checklist

- [ ] Tested changes against a test repository
- [ ] Added or updated relevant documentation (leave unchecked if not applicable)
- [ ] (If this PR is for a release) Updated README to point to the new tag (leave unchecked if not applicable)

### CODE REVIEW

File: src/main.ts [hunk 1]
```diff
@@ -182,6 +192,108 @@ class NixInstallerAction {
...
+      });
+    } catch (e) {
+      actions_core.debug(
+        `Did not detect \`/proc/self/cgroup\` existence, bailing on docker container ID detection:\n${e}`,
+      );
+      return false;
+    }
+
+    const cgroups = cgroups_buffer.trim().split("\n");
+    const last_cgroup = cgroups[cgroups.length - 1];
```
Comment (cole-h) [inline src/main.ts:217]: Would it be worth it to check _every_ cgroup line to see if there is `/docker/` in it, or if it's in an earlier cgroup, is that a weird configuration we wouldn't want to support anyways?

**Review by cole-h — APPROVED** [review]
I think this looks sound overall, but I'm not extremely familiar with docker or cgroups and the like, so take this approval with a grain of salt...

**Review by colemickens — APPROVED** [review]
Also no cgroup expert, but the code matches what was described, and the description I was given sounded convincing for detecting this scenario and avoiding it.

File: src/main.ts (hunk 1 above)
Comment (Hoverbear) [reply src/main.ts:217]: I don't have access to every system and setup, only a few, but I think in this case we're only looking for the `0::` prefixed line which is always last.

**Pull request merged**